packages/material-ui-utils/src/requirePropFactory.test.js

```diff
@@ -83,6 +84,26 @@ describe('requirePropFactory', () => {
           });
         });
       });
+
+      it('should chain the proptypes with the default prop types coming from the component', () => {
+        const Test = () => null;
+        const mock = spy();
+        Test.propTypes = {
+          test: mock,
+        };
+
+        const localProps = {};
+        const localPropName = 'test';
+        localProps[localPropName] = true;
+
+        const localRequireProp = requirePropFactory('Test', Test);
+
+        const result = localRequireProp('otherProp');
+        result(localProps, localPropName, undefined, undefined, undefined);
+
+        expect(mock.callCount).to.equal(1);
+        expect(mock.args[0]).to.deep.equal([localProps, localPropName, undefined, undefined, undefined]);
+      });
```

Comment:
Could you test these with `PropTypes.checkPropTypes` instead? Ideally you'd convert the other tests as well.

Reply:
@eps1lon can you check now if it is ok - https://github.com/mui-org/material-ui/pull/24423/commits/eb24b66a49ae459da05f8b673207fa41d584c087

If yes, I will update the other tests too.

Reply:
That looks way better :+1: I think you need to `PropTypes.resetWarningCache` in a `beforeEach` though. Otherwise subsequent runs in watchmode might fail. Not 100% certain though.

Reply:
Ok, will try it now after adding to all tests 👍

Reply:
Updated all tests, I believe it's ready for final review.